myreports/datasources/tjd_report.py

```
@@ -249,9 +253,14 @@ def filtered_query(self, company, filter_spec):
         filters = filter_spec
         filter_data = {}
         for k, f in filters.__dict__.iteritems():
-                formatted_filter = build_tjd_query(f)
-                filter_data[k] = formatted_filter
+            formatted_filter = build_tjd_query(f)
+            filter_data[k] = formatted_filter
         del filter_data['SentDate']
-        response = post_required_data(settings.TJD_ROOT_URL + 'report/report_a',
+
+        if isinstance(filter_data['Location'], dict):
+            filter_data['Location']['city'] = filter_data['Location'].get('city') or ''
+            filter_data['Location']['state'] = filter_data['Location'].get('state') or ''
+
+        response = post_required_data(settings.TJD_ROOT_URL + 'report/report_b',
```

If for any reason the response is 404, we will end up in an error `ValueError: No JSON object could be decoded`. This case needs to be handled

---

I agree. Since this issue wasn't introduced by this PR, I'd say that it isn't really a reason to stop this PR from being merged. On that note, I don't see anywhere that we gracefully handle runtime exceptions with reporting. We always return a 200, or, in the event of an error, a 500. So, we either return a default value when there's a non-JSON response (which hides the error more than handles it), or we create a workflow to return proper status codes from the run report fucntions. I'll create a ticket to handle different status codes coming out of reporting.

---

A ticket for that sounds good.